TAI Enhancements

scott-emery opened the pull request:
This is a fairly large modification to the TAI interface. It include a number
of modifications:

1. There are 3 APIs, and objects:
   a. Module - The optical module itself
   b. Network Interface - The optical interfaces leaving the module
   c. Host Interface - The internal interfaces connected to switching silicon
2. Module presence/absence indications from adapter to adapter host
3. Independence from SAI. All identifiers are changed from "SAI" to "TAI"
4. Documentation
   a. README.md describes the TAI interfaces/objects/attributes and their usage
   b. DOXYGEN comments for all APIs/objects/attributes
5. Removed SAI-specific functions which don't apply to TAI, like profiles.
6. Incorporate comments from pull request #1
   a. Included NTT's copyright on each file
   b. Added ifdef for proper C++ compilation
   c. Corrected misuse to "adapter" vs. "adapter host" in certain places.
   d. Removed all custom attributes. They can be added later.

Commit:
TAI Enhancements

This is a fairly large modification to the TAI interface. It include a number
of modifications:

1. There are 3 APIs, and objects:
   a. Module - The optical module itself
   b. Network Interface - The optical interfaces leaving the module
   c. Host Interface - The internal interfaces connected to switching silicon
2. Module presence/absence indications from adapter to adapter host
3. Independence from SAI. All identifiers are changed from "SAI" to "TAI"
4. Documentation
   a. README.md describes the TAI interfaces/objects/attributes and their usage
   b. DOXYGEN comments for all APIs/objects/attributes
5. Removed SAI-specific functions which don't apply to TAI, like profiles.

Signed-off-by: Scott Emery <scotte@cumulusnetworks.com>

Commit:
TAI Enhancements

This is a fairly large modification to the TAI interface. It include a number
of modifications:

1. There are 3 APIs, and objects:
   a. Module - The optical module itself
   b. Network Interface - The optical interfaces leaving the module
   c. Host Interface - The internal interfaces connected to switching silicon
2. Module presence/absence indications from adapter to adapter host
3. Independence from SAI. All identifiers are changed from "SAI" to "TAI"
4. Documentation
   a. README.md describes the TAI interfaces/objects/attributes and their usage
   b. DOXYGEN comments for all APIs/objects/attributes
5. Removed SAI-specific functions which don't apply to TAI, like profiles.
6. Incorporate comments from pull request #1
   a. Included NTT's copyright on each file
   b. Added ifdef for proper C++ compilation
   c. Corrected misuse to "adapter" vs. "adapter host" in certain places.
   d. Removed all custom attributes. They can be added later.

Signed-off-by: Scott Emery <scotte@cumulusnetworks.com>

Commit:
TAI Test Code

This patch modifies the test/test.c file to operate with the new TAI
enhancements submitted in the previous pull request. This include modifications
for the module presence/absence callback function, the module initialization,
network interface initialization and host interface initialization.

Signed-off-by: Scott Emery <scotte@cumulusnetworks.com>

ishidawataru commented:
Could you also edit `stub_tai.c` and make it compatible with your change?

https://github.com/Telecominfraproject/OOPT-DTC-TAI/blob/master/stub/stub_tai.c

scott-emery commented:
I was afraid you were going to ask for the stub_tai.c modifications too. I'll add that to this pull request in the next couple of days.

ishidawataru commented on README.md:
```diff
@@ -1,2 +1,289 @@
+
+The attr_list supplied on the `create_module()` function call must contain a 
+`TAI_MODULE_ATTR_LOCATION` attribute. This attribute defines which module is 
+being created. This is typically set the to value of the module_location 
```
set the to -> set to the

scott-emery replied:
Ack.

Commit:
TAI Stub Code

This patch modifies the stub/stub_tai.c file to operate with the new TAI
enhancements submitted in the previous pull request.

Signed-off-by: Scott Emery <scotte@cumulusnetworks.com>

Commit:
Update Makefile

Accidentally modified this file. Revert the modification.

ishidawataru commented:
Thanks!
